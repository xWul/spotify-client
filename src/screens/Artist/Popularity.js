import React from 'react'
import styled from 'styled-components'

const Label = styled.span`
  color: ${props => props.color ? props.color : '#333'};
  font-weight:  bold;
`

const Content = styled.p` 
  width: 100%;
`

const Popularity = ({ popularity }) => {
  let label = <Label>Underground</Label>
  if (popularity >= 80) {
    label = <Label color={'#FB5656'}>Hot</Label>
  } else if (popularity >= 60) {
    label = <Label color={'#EAEA00;'}>Cool</Label>
  } else if (popularity >= 30) {
    label = <Label color={'#7E7EE3'}>Regular</Label>
  }

  return <Content>{label}</Content>
}

export default Popularity
