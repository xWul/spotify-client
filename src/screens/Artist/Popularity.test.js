import React from 'react'
import renderer from 'react-test-renderer'

import Popularity from './Popularity'

describe('Popularity Artist', function () {
  test('Popularity Artist component renderer default', () => {
    const component = renderer.create(
      <Popularity popularity={0} />,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Popularity Artist component renderer default', () => {
    const component = renderer.create(
      <Popularity  popularity={90} />,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
