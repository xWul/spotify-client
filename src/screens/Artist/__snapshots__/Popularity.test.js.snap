// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popularity Artist Popularity Artist component renderer default 1`] = `
<p
  className="sc-bwzfXH jmLrRz"
>
  <span
    className="sc-bdVaJa cnqOZO"
  >
    Underground
  </span>
</p>
`;

exports[`Popularity Artist Popularity Artist component renderer default 2`] = `
<p
  className="sc-bwzfXH jmLrRz"
>
  <span
    className="sc-bdVaJa diIKEi"
    color="#FB5656"
  >
    Hot
  </span>
</p>
`;
